Automattic/wp-calypso
# Components: Refactor PlanStorage away from Reakit, remove Reakit dependency

## Proposed Changes

This PR refactors `PlanStorage` to the `Tooltip` from `@automattic/components` instead of using the Reakit one.

Allows us to get use from `reakit` as a dependency, which is going away from new version of `@wordpress/components` as well (in favor of `ariakit`).

This is also a blocker for https://github.com/Automattic/wp-calypso/pull/87350.

## Preview

This introduces some visual changes, which improve consistency since this is how the tooltip looks all across Calypso.

### Upgrade tooltip in `PlanStorage`

Before:
![Screenshot 2024-02-12 at 13 50 38](https://github.com/Automattic/wp-calypso/assets/8436925/e4a8d38e-b75e-49ae-924d-eb9caa9aa6f8)

After:
![Screenshot 2024-02-12 at 13 48 24](https://github.com/Automattic/wp-calypso/assets/8436925/e1268aa9-d59b-4819-85fb-b3ea072b482f)

### Shared quota tooltip in `PlanStorage:

Before:
![Screenshot 2024-02-12 at 13 52 23](https://github.com/Automattic/wp-calypso/assets/8436925/1aab18fc-2371-4db0-bfbc-e678a5e694ce)

After:
![Screenshot 2024-02-12 at 13 48 03](https://github.com/Automattic/wp-calypso/assets/8436925/37a12753-df58-48d0-982b-e4dcd986d9d4)

## Testing Instructions

* Go to `/media/:site` for:
  * A site that has a Creator plan
  * A site that has a Free plan
* Hover over the plan storage component on the right of the media library header.
* Verify the tooltip is there and looks good in both cases.
